ironfish/src/account/account.ts

```diff
@@ -162,21 +162,37 @@ export class Account {
     const blockHash = 'blockHash' in params ? params.blockHash : null
     let submittedSequence = 'submittedSequence' in params ? params.submittedSequence : null
 
+    let shouldUpdateTransaction = true
     const record = this.transactions.get(transactionHash)
     if (record) {
       submittedSequence = record.submittedSequence
+      shouldUpdateTransaction = !this.transactionValuesAreEqual(record, transaction, blockHash)
```

Comment:
nit: Seems like we could simplify code a bit by just 
```typescript
if (!record.transaction.equals(transaction) || !(record.blockHash === blockHash)) {
  await this.updateTransaction(
    transactionHash,
    { transaction, blockHash, submittedSequence },
    tx,
  )
}
```

Reply:
Yep sure, I can update it